doc/ansible/dev/dependent_watches.md

```diff
@@ -30,4 +30,117 @@ This is an example of a watches file with the `watchDependentResources` field se
+```
+
+#### Namespaced Resources
+
+For namespaced resources, we add an `ownerReference`.
+
+`$ kubectl edit deployment example-memcached-memcached`
+
+```yaml
+
+metadata:
+  ...(snip)
+  ownerReferences:
+    - apiVersion: cache.example.com/v1alpha1
+      kind: Memcached
+      name: example-memcached
+      uid: ad834522-d9a5-4841-beac-991ff3798c00
+```
+
+#### Cluster-scoped Resources
+
+For resources that are not namespaced, `ownerReferences` don't apply, so
+we need to add an `annotation`: (This namespace does not actually exist in the
+memcached example.)
+
+`$ kubectl edit namespace example-memcached`
+
+```yaml
+metadata:
+  annotations:
+    operator-sdk/primary-resource: default/example-memcached
+    operator-sdk/primary-resource-type: Memcached.cache.example.com
+```
+
+Warning: The value for primary-resource-type is "<kind>.<group>". Group
+must be separated from the `apiVersion` client-side. **If the `group` is a
+legacy string without a "/", the script below will fail.**
+
+#### Migration Script
+
+If you have many resources to update, you can modify the following
+**unsupported** script::
+
+```bash
+#!/usr/bin/env bash
+
+# This script is an unsupported example of how to migrate resources that
+# do not # have ownerReferences (or annotations for cluster-scoped
+# resources). Use at your own risk!  
+# https://github.com/operator-framework/operator-sdk/issues/1977
+
+# This script uses jq to parse JSON. https://stedolan.github.io/jq/
+
+# FIXME: Set this to an array of selfLinks of all resources to patch.
+declare -a resources_to_own=("/apis/apps/v1/namespaces/default/deployments/example-memcached-memcached" "/api/v1/namespaces/example-memcached")
+# FIXME: Set this to the name of the CR that should own the resources.
+local owning_cr_name="example-memcached"
+# FIXME: Set this to the name of the CRD that the CR belongs to.
+local crd_name="memcacheds.cache.example.com"
+
+
+owner_data=$(kubectl get $crd_name $owning_cr_name -o json)
+
+owner_reference=$(jq '. | {metadata: {ownerReferences: [{apiVersion:
+.apiVersion, kind: .kind, name: .metadata.name, uid: .metadata.uid}]}}' <<<"${owner_data}")
+
+owner_annotation=$(jq '. | {metadata:
+{annotations:{"operator-sdk/primary-resource": (.metadata.namespace + "/" +
+    .metadata.name), "operator-sdk/primary-resource-type": (.kind + "/" +
+    (.apiVersion | split("/")[0]))}}}' <<< "${owner_data}")
+
+echo "CR information: $owner_info"
+for resource in "${resources_to_own[@]}"
+do
+    resource_data=$(kubectl get --raw $resource)
+    type_name=$(jq -r  '.kind | ascii_downcase' <<< "${resource_data}")
+    name=$(jq -r  '.metadata.name' <<< "${resource_data}")
+    namespace=$(jq '.metadata | select(.namespace != null) .namespace' <<< "${resource_data}")
+    if [ -z "$namespace" ]
+    then
+        echo "$type_name $name is NOT a namespaced resource, adding owner info annotation"
+        kubectl patch $type_name $name -p $owner_annotation
```

Comment:
Is this patch idempotent, or if you run the script twice will the ownerReferences double up?